Add blogpost about changing translation downloads (#2136)
home-assistant/developers.home-assistant: blog/2024-04-05-translation-key-filtering.md

---
author: Joost Lekkerkerker
authorURL: https://github.com/joostlek
authorImageURL: https://avatars.githubusercontent.com/u/7083755?v=4
title: "Changing the way translation keys are handled"
---

> In Home Assistant Core 2024.4.2, we are changing the way translation keys are downloaded.

Reviewer: I found "downloaded" here is confusing. Download from where to where?
How about:
> In Home Assistant Core 2024.4.2, we are changing the way translations are included into builds from Lokalise.

> In Home Assistant Core 2024.4.2, we are changing the way translation keys are downloaded.

Reviewer 2: I found "downloaded" in this context is confusing. From where to where?

>  In Home Assistant Core 2024.4.2, we are changing the way translation are included into builds from Lokalise.

> Previously the script would download every translated string.

Reviewer: > Previously the build translation script would download every translated string.

> Previously the script would download every translated string.

Reviewer 2: > Previously the build translation script would download every translated string.

> With this change, we only download the strings that are verified.

Reviewer 2: > With this change, we only download the translations that are verified in Lokalise.

> A verified string is a translated string, and it becomes unverified when the original string changes.

Reviewer 3: I think this is better:
> All translations are verified when saved, but a translation becomes unverified id the default string is changed.

Author: ```suggestion
All translations are verified when saved, but a translation becomes unverified if the source string is changed
```

Reviewer 2: Since translation source is English:
> All translations are verified when saved, but a translation becomes unverified if the English string is changed.

> So whenever someone changes something in the original string, the translated string becomes unverified and needs to be reviewed again.

Reviewer 3: "something" is a pleonasm. "reviewed" is another Lokalize concept. Suggestions:
> So whenever someone changes the default value of a string (i.e. changes a `strings.json` file), all translations of that key becomes unverified and must be edited or marked as verified in Lokalise.

Reviewer: The "default" is English.

> So whenever the English string has been modified (i.e. changes a `strings.json` file), all translations of that key becomes unverified and must be edited or marked as verified in Lokalise.

Reviewer 2: > So whenever the English string is modified (i.e. changes a strings.json file), all translations which refers to the same translation key becomes unverified and must be edited or marked as verified in Lokalise.

> This is very beneficial for the translation keys that make use of translation key placeholders, since they can now never have a mismatch between the original and translated string.

Reviewer 3: I think this is better:
> This catches both significant changes to strings that may be hidden if the translation is not updated, and also avoids placeholder mismatches. Note that correcting typos in the default strings also will make translations "unverified".

Reviewer: > This catches changes to English strings that may be hidden if the translation is not updated, and also avoids placeholder mismatches. Note that correcting typos in the English strings also will make translations "unverified".

Reviewer 2: > This catches changes to English strings that may be hidden if the translation is not updated, and also avoids placeholder mismatches. Note that correcting typos in the English strings also will make translations "unverified".

Previously the script would download every translated string.
With this change, we only download the strings that are verified.

> All translations are verified when saved, but a translation becomes unverified if the source string is changed

Reviewer 3: Nit: missing punctuation at end of sentence.
